ion/unmarshal.go

```diff
@@ -196,9 +199,15 @@ func (d *Decoder) decodeInt() (interface{}, error) {
 		return nil, nil
 	case Int32:
 		val, err := d.r.IntValue()
+		if err != nil {
+			return nil, err
+		}
 		return *val, err
 	case Int64:
 		val, err := d.r.Int64Value()
+		if err != nil {
+			return nil, err
+		}
 		return *val, err
 	default:
 		return d.r.BigIntValue()
```

Comment:
Could you update Unmarshal API to support shared symbol tables.

I was playing around marshalling/ unmarshalling roundtrip with sst and it was missing.

Update L95-97 to:
```
func Unmarshal(data []byte, v interface{}, ssts ...SharedSymbolTable ) error {
	catalog := NewCatalog(ssts...)
	return NewDecoder(NewReaderCat(bytes.NewReader(data), catalog)).DecodeTo(v)
}
```

Reply:
Addressed.